backend/controllers/todoList.controller.js

```diff
@@ -1,68 +1,149 @@
 const Todo = require("../models/todoList.model");
+const { commonErrorHandler } = require("../helper/errorHandler.helper");
+const { asyncHandler } = require("../middlewares/asyncHandler.middleware");
 
-const createItem = async(req, res) => {
-  const body = req.body;
+// @desc    Add Todo Item
+// @route   POST /api/todo-list
+// @access  Private
 
-  const todo = new Todo(body);
-  const data = await todo.save();
-  if(!data){
-    return res.status(400).json({
-      error:"Item cannot be saved"
-    })
+const createTodoItem = asyncHandler(async (req, res) => {
+  try {
+    const body = req.body;
+
+    // create a schema todo item
+    const todo = new Todo(body);
+
+    // if something in syntax goes wrong
+    if (!todo) {
+      throw new Error();
+    }
+
+    // save todo item in DB
+    const data = await todo.save();
+    if (!data) {
+      throw new Error();
+    }
+
+    // final response
+    return commonErrorHandler(
+      req,
+      res,
+      { data, quote: "Item has been created" },
+      201
+    );
+  } catch (error) {
+    return commonErrorHandler(req, res, null, 500, error);
   }
-  return res.status(201).json(data);
+});
 
-};
+// @desc    Get All Todo Items
+// @route   GET  /api/todo-list
+// @access  Private
+
+// @desc    Get All Todo Items By Category
+// @route   GET  /api/todo-list?category=?
+// @access  Private
 
-const getAllItems = async (req, res) => {
-  const query = req.query;
-  if (Object.keys(req.query).length === 0) {
-    const data = await Todo.find({});
-    return res.status(200).json(data);
+const getAllTodoItems = asyncHandler(async (req, res) => {
+  try {
+    const query = {};
+
+    for (const key in req.query) {
```

Comment:
why is this required to get all todos and what about if other user tried to fetch todos of other user by id or title